src/App.tsx

```diff
@@ -1,5 +1,64 @@
-const App = () => {
-    return <h1>Hello World</h1>;
-};
+// Importing React from the React package
+// Importing colors from the our file path ./colors.json
+import React, { useState } from 'react';
+import colors from './colors.json';
 
-export default App;
+//Interface Color and what you expect, hex, name, and comp
+interface Color {
+  hex: string;
+  name: string;
+  comp: { hex: string; name: string }[];
+}
+
+//Create a state variable that will give us our main color first
+//State variable gets upadted and rerendered 
+const ColorCard: React.FC<Color> = ({ hex, name, comp }) => {
+    const [displayHex, setDisplayHex] = useState(hex);
+
+
+//Define Onmouse what happens when a user hovers over the element
+//When the mouse enters the div, it calls displayHex and rerenders the component
+    return (
+      <div className="m-4 p-4 border rounded-lg shadow-md">
+        <div className="flex items-center mb-2">
+          <div 
+            className="w-12 h-12 rounded-full mr-4" 
+            style={{ backgroundColor: `#${hex}` }}
+            onMouseEnter={() => setDisplayHex(hex)}
+          ></div>
+          <h2 className="text-xl font-bold">{name}</h2>
+        </div>
+        <p className="text-gray-600 mb-2">Hex: #{displayHex}</p>
+        <div className="flex flex-wrap">
+          {comp.map((c, index) => (
+    // When you move your mouse over a complementary color swatch,
```

Comment:
Great to leave comments describing code, but I would consider trying to make this more concise.